src/main/java/org/cyclonedx/gradle/model/SbomComponent.java

```diff
@@ -0,0 +1,125 @@
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.apache.maven.model.License;
+
+public final class SbomComponent implements Serializable {
+
+    private final SbomComponentId id;
+    private final Set<ConfigurationScope> inScopeConfigurations;
+    private final Set<SbomComponentId> dependencyComponents;
+
+    @Nullable private final File artifactFile;
+
+    @Nullable private final SbomMetaData metaData;
+
+    @Nullable private final List<License> licenses;
+
+    public SbomComponent(
+            final SbomComponentId id,
+            final Set<ConfigurationScope> inScopeConfigurations,
+            final Set<SbomComponentId> dependencyComponents,
+            @Nullable final File artifactFile,
+            @Nullable final SbomMetaData metaData,
+            @Nullable final List<License> licenses) {
+        this.id = id;
+        this.inScopeConfigurations = inScopeConfigurations;
+        this.dependencyComponents = dependencyComponents;
+        this.artifactFile = artifactFile;
+        this.metaData = metaData;
+        this.licenses = licenses;
+    }
+
+    public SbomComponentId getId() {
+        return id;
+    }
+
+    public Set<ConfigurationScope> getInScopeConfigurations() {
+        return inScopeConfigurations;
+    }
+
+    public Set<SbomComponentId> getDependencyComponents() {
+        return dependencyComponents;
+    }
+
+    public Optional<File> getArtifactFile() {
+        return Optional.ofNullable(artifactFile);
+    }
+
+    public Optional<SbomMetaData> getSbomMetaData() {
+        return Optional.ofNullable(metaData);
+    }
+
+    public Optional<List<License>> getLicenses() {
+        return Optional.ofNullable(licenses);
+    }
+
+    public static class Builder {
+
+        private SbomComponentId id;
+        private Set<ConfigurationScope> inScopeConfigurations;
+        private Set<SbomComponentId> dependencyComponents;
+        private File artifactFile;
+        private SbomMetaData metaData;
+        private List<License> licenses;
+
+        public Builder() {}
+
+        public Builder withId(final SbomComponentId id) {
+            this.id = id;
+            return this;
+        }
+
+        public Builder withInScopeConfigurations(final Set<ConfigurationScope> inScopeConfigurations) {
+            this.inScopeConfigurations = inScopeConfigurations;
+            return this;
+        }
+
+        public Builder withDependencyComponents(final Set<SbomComponentId> dependencyComponents) {
+            this.dependencyComponents = dependencyComponents;
+            return this;
+        }
+
+        public Builder withArtifactFile(@Nullable final File artifactFile) {
+            this.artifactFile = artifactFile;
+            return this;
+        }
+
+        public Builder withMetaData(@Nullable final SbomMetaData metaData) {
+            this.metaData = metaData;
+            return this;
+        }
+
+        public Builder withLicenses(@Nullable final List<License> licenses) {
```

Comment:
Please use empty list here instead of nullable collection

Reply:
Done